libsolidity/ast/AST.cpp

```diff
@@ -151,7 +151,7 @@ bool ContractDefinition::constructorIsPublic() const
 
 bool ContractDefinition::canBeDeployed() const
 {
-	return constructorIsPublic() && annotation().unimplementedFunctions.empty();
+	return constructorIsPublic() && annotation().unimplementedFunctions.empty() && !abstract();
```

Comment:
```suggestion
	return constructorIsPublic() && !abstract();
```

Reply:
Looks like that `annotation().unimplementedFunctions.empty() ` is needed here, if an interface is used.

Reply:
The method `abstract()` only returns whether a contract was marked as abstract. So if e.g. an interface was used, there could be still unimplemented functions.

Reply:
Sure, but that will be caught by a different error elsewhere.